src/newProject/NewProjectPanel.ts

```diff
@@ -0,0 +1,414 @@
+        openOcdConfigFilesList,
+      });
+      const idfTarget = idfConf.readParameter("idf.adapterTargetName");
+      this.panel.webview.postMessage({
+        command: "load_adapter_target",
+        idfTarget,
+      });
+    } catch (error) {
+      OutputChannel.appendLine(error);
+      Logger.errorNotify("Error loading initial values in New Project", error);
+    }
+  }
+
+  public async loadToolsVenvForIdf(
+    metadata: IMetadataFile,
+    selectedIdf: IPath
+  ) {
+    this.panel.webview.postMessage({
+      command: "load_idf_versions",
+      idfVersions: metadata.idf,
+      selectedIdf,
+    });
+    const idfVersion = await utils.getEspIdfVersion(selectedIdf.path);
+    this.panel.webview.postMessage({
+      command: "load_current_idf_version",
+      idfVersion,
+    });
+    const pyVenvList = metadata.venv.filter(
+      (venv) => venv.path.indexOf(idfVersion) > -1
+    );
+    const pyBinPath = idfConf.readParameter("idf.pythonBinPath") as string;
+    const selectedVenv =
+      pyVenvList.find((venv) => venv.path === pyBinPath) || pyVenvList[0];
+    this.panel.webview.postMessage({
+      command: "load_py_venvs",
+      pyVenvList,
+      selectedVenv,
+    });
+    const toolsList = await getToolsInMetadataForIdfPath(
+      selectedIdf.path,
+      metadata.tools
+    );
+    this.panel.webview.postMessage({
+      command: "load_tools_list",
+      tools: toolsList,
+    });
+  }
+
+  public async checkSettings(idf: IPath, tools: ITool[], venv: IPath) {
+    const toolsResult = await utils.validateToolsFromMetadata(idf.path, tools);
+    const idfPyReqs = join(idf.path, "requirements.txt");
+    const debugAdapterPyReqs = join(
+      this.extensionPath,
+      "esp_debug_adapter",
+      "requirements.txt"
+    );
+    let allIsValid: boolean = true;
+    for (const tKey of Object.keys(toolsResult)) {
+      if (!toolsResult[tKey]) {
+        allIsValid = false;
+        Logger.infoNotify(`Bin path is not valid for ${tKey}`);
+        OutputChannel.appendLine(`Bin path is not valid for ${tKey}`);
+      }
+    }
+    const idfPyLog = await utils.startPythonReqsProcess(
+      venv.path,
+      idf.path,
+      idfPyReqs
+    );
+    const resultLog = `Checking Python requirements using ${venv.path}\n${idfPyLog}`;
+    OutputChannel.appendLine(resultLog);
+    Logger.info(resultLog);
+    const debugPyLog = await utils.startPythonReqsProcess(
+      venv.path,
+      idf.path,
+      debugAdapterPyReqs
+    );
+    const adapterResultLog = `Checking Debug Adapter requirements using ${venv.path}\n${debugPyLog}`;
+    OutputChannel.appendLine(adapterResultLog);
+    Logger.info(adapterResultLog);
+    if (
+      idfPyLog.indexOf("are not satisfied") > -1 &&
+      debugPyLog.indexOf("are not satisfied") > -1
+    ) {
+      allIsValid = false;
+    }
+    this.panel.webview.postMessage({
+      command: "idf_tools_check_done",
+      allIsValid,
+    });
+  }
+
+  public async createProject(
```

Comment:
call this inside a `vscode.window.withProgress(...)` to show a progress bar to the user.